src/shared/compatibility.js

```diff
@@ -1015,6 +1020,11 @@ PDFJS.compatibilityChecked = true;
       }
       return this._scheme + '://' + host;
     },
+
+    // Issue 8726: Safari does not fallback to the original URL object.
+    get searchParams() {
+      return new URLSearchParams(this.search);
```

Comment:
Also, given https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams#Browser_compatibility, I'm not sure how this helps in general considering that this won't work in other browsers that need to `URL` polyfill (e.g. IE11).